MarsRover/Rover.cs

```diff
@@ -101,5 +96,10 @@ private void MoveBackward(int maxWidth, int maxHeight)
                 _ => new Location(x, y)
             };
         }
+
+        public Location GetTargetLocation(Command command, int maxWidth, int maxHeight)
+        {
+            return command == Command.Forward ? MoveForward(maxWidth, maxHeight) : MoveBackward(maxWidth, maxHeight);

```

Comment:
MoveForward and MoveBackward actually changes the location of the rover. If you just want to check the target location without changing the actual location of the Rover you can create a new method and repeat some of the code that's in MoveForward and MoveBackward to calculate the target location, but don't set the Location.